timing-entrytypes-registry/registry.html

```diff
@@ -0,0 +1,91 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8" />
+<title>Registry test for Performance Timeline</title>
+<link rel="help" href="https://w3c.github.io/timing-entrytypes-registry/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+const STEPS = {};
+
+const types = (self.PerformanceObserver
+                  && self.PerformanceObserver.supportedEntryTypes)?
+    self.PerformanceObserver.supportedEntryTypes
+    : undefined;
+
+if (types) {
+  // we observe everything as soon as possible
+  new PerformanceObserver(function (list, observer) {
+    for (const entry of list.getEntries())
+      if (STEPS[entry.entryType]) STEPS[entry.entryType](entry);
+  }).observe({entryTypes: self.PerformanceObserver.supportedEntryTypes});
+}
+
+function test_support(def) {
+  if (!types || !types.includes(def[0])) {
+    return;
+  }
+  const desc = `'${def[0]}' entries should be observable`;
+  const t = async_test(desc);
+
+  STEPS[def[0]] = (entry) => {
+    t.step(() => assert_equals(Object.prototype.toString.call(entry),

```

Comment:
Hmm is there a cleaner way to do this?

Reply:
it's actually pretty clean I think. It's listening to all supported entries by default so one doesn't have to worry about it when adding a new entry.